rclc/include/rclc/executor.h

```diff
@@ -50,14 +50,26 @@ typedef enum
   LET
 } rclc_executor_semantics_t;
 
+typedef enum
+{
+  NONE,
+  SINGLE_THREADED,
+  MULTI_THREADED,
+  NON_POSIX,
+} rclc_executor_type_t;
+
 /// Type definition for trigger function. With the parameters:
 /// - array of executor_handles
 /// - size of array
 /// - application specific struct used in the trigger function
 typedef bool (* rclc_executor_trigger_t)(rclc_executor_handle_t *, unsigned int, void *);
 
+/// function pointer specification
+typedef struct rclc_executor_t_ rclc_executor_t;
+typedef rcl_ret_t (* rclc_executor_func_t)(rclc_executor_t *);
+
 /// Container for RCLC-Executor
-typedef struct
+struct rclc_executor_t_
```

Comment:
Is the last `_` intended? Usually in ROS 2 they use this approach: https://github.com/ros2/rmw/blob/e523e4706ae6e20134fb835a91d18b74a134175e/rmw/include/rmw/discovery_options.h#L48-L51

The typedef and the structure definition are onelined and the structure subfix is `_s`. 

If you agree I prefer to follow this approach.

Reply:
That's a good suggestion, Pablo. 
In the multi-threaded executor PR I also wanted to have the function pointers in the API interface, like here: 
For that I need two lines (as forward definition).

https://github.com/ros2/rclc/pull/339/files#diff-2b97ef21c9b9eb498bff3a8723efb6469f26889ffd0ab32e5b6e88ca2594abcaR103
```
typedef struct rclc_executor_t_ rclc_executor_t;
typedef rcl_ret_t (* rclc_executor_func_t)(rclc_executor_t *);
```
```
  void * custom;

  /// interface functions
  rclc_executor_func_t init;
  rclc_executor_func_t spin_init;
};

Then I thought, because I don't know yet, how many functions do I need, I'll hide them also in the `custom` pointer. So the data structure does not need to be modified. 

But I just wanted to have the typedef definitions already defined.